controllers/etcd_controller.go

```diff
@@ -179,11 +180,29 @@ func (r *EtcdReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	if !reflect.DeepEqual(etcd.Spec, etcdCopy.Spec) {
 		etcdCopy.Spec = etcd.Spec
 	}
-	logger.Infof("Reconciling etcd: %s/%s", etcd.GetNamespace(), etcd.GetName())
+
+	etcdCopy.Status.ObservedGeneration = &etcdCopy.Generation
+	if _, ok := etcd.Annotations[common.GardenerOperation]; ok {
+		delete(etcd.Annotations, common.GardenerOperation)
+		if err := r.Update(context.TODO(), etcd); err != nil {
+			if err := r.updateEtcdErrorStatus(etcd, err); err != nil {
+				return ctrl.Result{
+					Requeue:      true,
+					RequeueAfter: time.Second * 5,
+				}, err
+			}
+			return ctrl.Result{
+				Requeue:      true,
+				RequeueAfter: time.Second * 5,
```

Comment:
Why is this needed? It automatically requeues with exponential backoff on error.

Reply:
makes sense.